File: Sources/iTunesKit/Core/iTunesSearchResponse.swift
//
//  iTunesSearchResponse.swift
//  iTunesKit
//
//  Created by Aarav Gupta on 21/03/26.
//

import Foundation

// MARK: - iTunes Search API Models
public struct iTunesSearchResponse: Codable, Sendable {
    public let resultCount: Int
    public let results: [iTunesSearchResult]
}

public struct iTunesSearchResult: Codable, Sendable, Hashable {
    public let wrapperType: String?
    public let kind: String?
    public let artistId: Int?
    public let collectionId: Int?
    public let trackId: Int?
    public let artistName: String?
    public let collectionName: String?
    public let trackName: String?
    public let trackViewUrl: String?
    public let collectionViewUrl: String?
    public let previewUrl: String?
    public let artworkUrl30: String?
    public let artworkUrl60: String?
    public let artworkUrl100: String?
    public let releaseDate: String?
    public let country: String?
    public let currency: String?
    public let trackExplicitness: String?
    public let collectionExplicitness: String?
    public let contentAdvisoryRating: String?
    public let primaryGenreName: String?
}

File: ituneskit/src/main/java/aaravgupta/ituneskit/core/iTunesSearchResponse.kt
package aaravgupta.ituneskit

import kotlinx.serialization.Serializable

@Serializable
public data class iTunesSearchResponse(
    val resultCount: Int,
    val results: List<iTunesSearchResult>
)

@Serializable
public data class iTunesSearchResult(
    val wrapperType: String? = null,
    val kind: String? = null,
    val artistId: Int? = null,
    val collectionId: Int? = null,
    val trackId: Int? = null,
    val artistName: String? = null,
    val collectionName: String? = null,
    val trackName: String? = null,
    val trackViewUrl: String? = null,
    val collectionViewUrl: String? = null,
    val previewUrl: String? = null,
    val artworkUrl30: String? = null,
    val artworkUrl60: String? = null,
    val artworkUrl100: String? = null,
    val releaseDate: String? = null,
    val country: String? = null,
    val currency: String? = null,
    val trackExplicitness: String? = null,
    val collectionExplicitness: String? = null,
    val contentAdvisoryRating: String? = null,
    val primaryGenreName: String? = null
)